tests/e2e/performance/csi_tests/test_pvc_clone_performance.py

```diff
@@ -203,4 +251,80 @@ def test_clone_create_delete_performance(
+        # Collecting environment information
+        self.get_env_info()
+
+        self.full_log_path = get_full_test_logs_path(cname=self)
+        self.results_path = get_full_test_logs_path(cname=self)
+        self.full_log_path += f"--PVC-SIZE-{pvc_size}"
+        logger.info(f"Logs file path name is : {self.full_log_path}")
+
+        # Initialize the results doc file.
+        full_results = self.init_full_results(
+            ResultsAnalyse(
+                self.uuid,
+                self.crd_data,
+                self.full_log_path,
+                "pvc_clone_performance",
+            )
+        )
+
+        full_results.add_key("interface", self.interface)
+        full_results.add_key("total_clone_number", max_num_of_clones)
+        full_results.add_key("pvc_size", self.pvc_size)
+        full_results.add_key("clone_creation_time", creation_time_list)
+        full_results.add_key("clone_deletion_time", deletion_time_list)
+        full_results.add_key("average_clone_creation_time", average_creation_time)
+        full_results.add_key("average_clone_deletion_time", average_deletion_time)
+        full_results.add_key("clone_creation_speed", creation_speed_list)
+        full_results.add_key("clone_deletion_speed", deletion_speed_list)
+        full_results.add_key("average_clone_creation_speed", average_creation_speed)
+        full_results.add_key("average_clone_deletion_speed", average_deletion_speed)
+
+        # Write the test results into the ES server
+        if full_results.es_write():
+            res_link = full_results.results_link()
+            logger.info(f"The Result can be found at : {res_link}")
+
+            # Create text file with results of all subtest (4 - according to the parameters)
+            self.write_result_to_file(res_link)
+
+    def test_pvc_clone_results(self):
+        """
+        This is not a test - it is only check that previous test ran and finish as expected
+        and reporting the full results (links in the ES) of previous tests (4)
+        """
+        self.number_of_tests = 8
+        self.results_path = get_full_test_logs_path(
+            cname=self, fname="test_clone_create_delete_performance"
+        )
+        self.results_file = os.path.join(self.results_path, "all_results.txt")
+        logger.info(f"Check results in {self.results_file}.")
+        self.check_tests_results()
+        self.push_to_dashboard(test_name="PVC Clone Performance")
+
+    def init_full_results(self, full_results):
+        """
+        Initialize the full results object which will send to the ES server
+
+        Args:
+            full_results (obj): an FIOResultsAnalyse object
+
+        Returns:
+            FIOResultsAnalyse (obj): the input object fill with data
+
+        """
+        for key in self.environment:
+            full_results.add_key(key, self.environment[key])
+        full_results.add_key("index", full_results.new_index)
+        return full_results
```

Comment:
Can this function be moved to generic library file?

Reply:
@Avilir @ypersky1980 @pintojoy ^^ if its more specific to test-case then ok with it but all the duplicate function do have same lines of codes.

Reply:
@pintojoy @ramkiperiy I think that we have to have this def init_full_results(self, full_results) in each one of our tests - this is initialization of the results obj.

Reply:
we do have a generic class ResultsAnalyse(PerfResult): in ocs-ci/ocs_ci/ocs/perfresult.py /

https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/perfresult.py#L163

Reply:
@ypersky1980 yes, even my point was same why can't we add this function `init_full_results` in the class `class ResultsAnalyse(PerfResult):` and call it in each test script ? 
@Avilir WDYT about this ?

Reply:
@ramkiperiy , the reason for this object is to hendale the results, calculating average , summarize results etc.

Reply:
As discussed in the meeting, let's have this tracked part of code optimization and reducing code duplicates.